aries-test-harness/agent_test_utils.py

```diff
@@ -52,3 +52,33 @@ def get_relative_timestamp_to_epoch(timestamp):
 
     return epoch_time
 
+def format_cred_proposal_by_aip_version(context, aip_version, cred_data, connection_id, filters=None):
+
+    if aip_version == "AIP20":
+
+        filters = amend_filters_with_runtime_data(context, filters)
+
+        credential_proposal = {
+            "connection_id": connection_id,
+            "credential_preview": {
+                "@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/issue-credential/1.0/credential-preview",
```

Comment:
Should be didcomm.org + 2.0 right?

```suggestion
                "@type": "https://didcomm.org/issue-credential/2.0/credential-preview",
```

Reply:
Yes, true about @type. Thanks. 

Right now the tests are structured to only do one format per scenario. I think that is a decent start. Using the tags, a framework could construct a set of tests they support. After we have the single format for a single scenario working well. If won't be hard to create scenarios that allow multiple format tags and will allow to combine in one scenario. 

For the proposal and offer scenario you mention, we should be able write the steps you are talking about like this. 
```
@T00X-RFC0453 @critical @AcceptanceTest @DIDExchangeConnection @CredFormat_Indy @CredFormat_JSON-LD @Schema_DriversLicense_v2
  Scenario Outline: Issue a Indy credential with the Holder beginning with a proposal of a indy and json-ld credential
    Given "2" agents
      | name | role   |
      | Acme | issuer |
      | Bob  | holder |
    And "Acme" and "Bob" have an existing connection
    When "Bob" proposes a "indy + json-ld" credential to "Acme" with <credential_data>
    And "Acme" offers an "indy" credential with <different_credential_data>
    And "Bob" requests the "indy" credential
    And "Acme" issues the "indy" credential
    And "Bob" acknowledges the "indy" credential issue
    Then "Bob" has the "indy" credential issued
```

Not sure if that answers all your questions or not? I don't know what the signature algorithm means, is that foreign to the RFC? If needed to differentiate, we will have to add it to the Scenario tag(s), or integrate it into the gherkin step definition.